### PR TITLE
Keep a single null dimension value

### LINKED ISSUE #39602: [dc.js migration] should render a single null dimension value
question/12561
https://metaboat.slack.com/archives/C06G94JTWBS/p1706197339520639

### PR DESCRIPTION
**Pull request by kulyk**
Closes #39602

Updates the dataset transform logic to throw an error if we end up with an empty dataset after filtering null-ish dimension values

### Demo

CleanShot 2024-05-30 at 14 27 27@2x

### CODE REVIEW

**Review by alxnddr — COMMENTED** [review]
@kulyk @cdeweyx I think for users it would be better if we show an error message that tells there is no data to display, wdyt?

**Comment by cdeweyx** [comment]
> @kulyk @cdeweyx I think for users it would be better if we show an error message that tells there is no data to display, wdyt?

Yeah agreed

**Comment by kulyk** [comment]
@alxnddr, thanks for bringing this up; I've updated the PR to throw an error in this case. React looses the error message in dev mode, but it works fine in production builds. I attached a screenshot from a locally built uberjar with a demo

File: frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts [hunk 1]
```diff
@@ -622,6 +622,11 @@ export const applyVisualizationSettingsDataTransformations = (
     xAxisModel.isHistogram
   ) {
     dataset = filterNullDimensionValues(dataset, showWarning);
+    if (dataset.length === 0) {
+      throw new Error(
+        t`There is no data to display. Check the query to ensure there are non-null x-axis values.`,
```
Comment (EmmadUsmani) [inline frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts:627]: Nit: we can move the string into a constant to reuse it in the unit test

**Review by EmmadUsmani — APPROVED** [review]

**Comment by EmmadUsmani** [comment]
Looks good! Let's run the copy by @cdeweyx and @kdoh just to see if we should tweak the wording before merging

**Comment by kulyk** [comment]
> Looks good! Let's run the copy by @cdeweyx and @kdoh just to see if we should tweak the wording before merging

fwiw Conor came up with the error message, so it should be goot to go 🚢

**Pull request merged**